### PR TITLE
Clean up the Icinga plugins a bit

### PR DESCRIPTION
**Pull request by gunnarbeutner**
This cleans up a _lot_ of code style issues with our Windows plugins.

### CODE REVIEW

**Comment by dnsmichi** [comment]
From a quick look, this is a total rewrite. I'll wait until all todos are marked :)

**Comment by gunnarbeutner** [comment]
Well, no, it's not a rewrite. Just a whole _ton_ of refactoring. 😇

**Comment by Crunsher** [comment]
>Add namespace (icinga?)
Please don't, except for check_nscp the plugins are stand alone. I would like to keep it that way and possibly move them out of the project into their own in the future.

**Comment by gunnarbeutner** [comment]
Seems reasonable enough, considering, well... adding namespace {} around a single .cpp file does appear to be rather silly. 😇

File: plugins/check_perfmon.cpp [hunk 1]
```diff
@@ -314,27 +304,26 @@ BOOL QueryPerfData(printInfoStruct& pI)
...
 	if (FAILED(status))
 		goto die;
 
-	status = PdhGetFormattedCounterArray(hCounter, pI.dwRequestedType, &dwBufferSize, &dwItemCount, pDisplayValues);
+	status = PdhGetFormattedCounterArray(hCounter, pI.dwRequestedType, &dwBufferSize, &dwItemCount, NULL);
 	if (status != PDH_MORE_DATA)
 		goto die;
 
-	pDisplayValues = reinterpret_cast<PDH_FMT_COUNTERVALUE_ITEM*>(new BYTE[dwBufferSize]);
+	PDH_FMT_COUNTERVALUE_ITEM *pDisplayValues = reinterpret_cast<PDH_FMT_COUNTERVALUE_ITEM*>(new BYTE[dwBufferSize]);
```
Comment (Crunsher) [inline plugins/check_perfmon.cpp]: pDisplayValues needs to be declared before we can run into a `goto die` as it is referenced there.

File: plugins/check_network.cpp [hunk 2]
```diff
@@ -211,77 +199,82 @@ INT printOutput(printInfoStruct& printInfo, CONST std::vector<nInterface>& vInte
...
 	if (!SUCCEEDED(err))
 		goto die;
 
-	if (debug)
+	if (l_Debug)
 		std::wcout << L"Creating formatted counter arrays" << '\n';
 
+	DWORD dwItemCount;
+	DWORD dwBufferSizeIn = 0;
+	PDH_FMT_COUNTERVALUE_ITEM *pDisplayValuesIn = NULL, *pDisplayValuesOut = NULL;
```
Comment (Crunsher) [inline plugins/check_network.cpp]: This needs to happen before any goto statements.

File: plugins/check_service.cpp [hunk 3]
```diff
@@ -195,7 +176,7 @@ std::wstring GetServiceByDescription(CONST std::wstring& description) {
 
 	LPENUM_SERVICE_STATUSW lpServices = reinterpret_cast<LPENUM_SERVICE_STATUSW>(new BYTE[pcbBytesNeeded]);
```
Comment (Crunsher) [inline plugins/check_service.cpp]: Also needs to be declared before the first goto

**Comment by gunnarbeutner** [comment]
Oops, should be fixed now. :P

**Comment by Crunsher** [comment]
check_service still has that problem. It wasn't introduced by your changes but there is no need to have a separate issue for this.

**Comment by MaBauMeBad** [comment]
If you are "rewriteing" the Plugins now, could you please implement a function for check_proc like on linux to check a single process ?

Otherwise i would open a new Request for that.

Thanks ;)

**Comment by dnsmichi** [comment]
This PR is about code quality, not for adding any new features.

**Comment by gunnarbeutner** [comment]
Fixed. There are also a few other issues which I decided not to fix in this PR:

* mismatching delete:

```LPENUM_SERVICE_STATUSW lpServices = NULL;
lpServices = reinterpret_cast<LPENUM_SERVICE_STATUSW>(new BYTE[pcbBytesNeeded]);
delete[] lpServices;```

(This should be `delete [] reinterpret_cast<BYTE *>(lpServices)`.)

* There are far too many reinterpret_casts, news and deletes to begin with.

* Some of the variable, struct and field names could use an overhaul: tRam, aRam, Bunit, etc.

**Comment by dnsmichi** [comment]
As learned from https://help.github.com/articles/checking-out-pull-requests-locally/

```
C:\Users\michi\source\repos\icinga2>git fetch origin pull/6058/head:feature/clean-up-plugins
remote: Counting objects: 19, done.
remote: Total 19 (delta 18), reused 18 (delta 18), pack-reused 1
Unpacking objects: 100% (19/19), done.
From ssh://github.com/icinga/icinga2
 * [new ref]             refs/pull/6058/head -> feature/clean-up-plugins

C:\Users\michi\source\repos\icinga2>git checkout feature/clean-up-plugins
Switched to branch 'feature/clean-up-plugins'
```

**Comment by Crunsher** [comment]
AppVeyor fails with something that looks related to #6109

**Comment by dnsmichi** [comment]
Yep, tried to build the PR yesterday, check_memory is broken. Will look into a fix when I find the time.

**Pull request merged**